src/main/java/in/projecteka/consentmanager/SecurityConfiguration.java

```diff
@@ -174,14 +174,13 @@ private boolean isEmpty(String authToken) {
         }
 
         private Mono<SecurityContext> checkSignUp(String authToken) {
-            if (!signupService.validateToken(authToken)) {
-                return Mono.empty();
-            }
-            return Mono.just(new UsernamePasswordAuthenticationToken(
-                    authToken,
-                    authToken,
-                    new ArrayList<SimpleGrantedAuthority>()))
-                    .map(SecurityContextImpl::new);
+            return Mono.just(authToken)
+                    .filterWhen(token -> signupService.validateToken(token))
```

Comment:
can't we do? 
```
signupService.validateToken(token)
.flatMap(token -> Mono.just(new UsernamePasswordAuthenticationToken(
                            token,
                            token,
                            new ArrayList<SimpleGrantedAuthority>()))
                            .map(SecurityContextImpl::new))
```

Reply:
I tried to retain `signupService.validateToken` to return Boolean(Wrapped in Mono). So `filterwhen` comes in nice to pass through the actual value from Publisher.

Reply:
I don't get retaining boolean, but I'm pretty sure that you don't need to do Mono.just(authToken) at all, which is not a good practise when you can pass it.